eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -96,8 +96,9 @@ object OnionMessages {
     } else {
       payloadSize.toInt
     }
-    val Sphinx.PacketAndSecrets(packet, _) = Sphinx.create(sessionKey, packetSize, route.blindedNodes.map(_.blindedPublicKey), payloads, None)
```

Comment:
Here we specifically choose a `packetSize` that fits the payload. Could `Sphinx.create` still fail?

Reply:
You're right, I don't think it can fail in that case (or if it does it really is a bug that should be fixed asap, so it's better to let it crash). That will simplify the `OnionMessages` part :+1:

Reply:
I changed to a simple `.get` and a comment in https://github.com/ACINQ/eclair/pull/2087/commits/4c9b77c1045ef437deb40f9c4c869a3065a93dc1